packages/rancher-monitoring/rancher-monitoring.patch

```diff
@@ -193,19 +193,49 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/cha
      {{- end }}
      imagePullPolicy: {{ .Values.image.pullPolicy }}
    {{- if .Values.command }}
+diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/charts/grafana/templates/podsecuritypolicy.yaml packages/rancher-monitoring/charts/charts/grafana/templates/podsecuritypolicy.yaml
+--- packages/rancher-monitoring/charts-original/charts/grafana/templates/podsecuritypolicy.yaml
++++ packages/rancher-monitoring/charts/charts/grafana/templates/podsecuritypolicy.yaml
+@@ -6,13 +6,9 @@
+   namespace: {{ template "grafana.namespace" . }}
+   labels:
+     {{- include "grafana.labels" . | nindent 4 }}
+-  annotations:
+-    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
+-    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
+-    {{- if .Values.rbac.pspUseAppArmor }}
+-    apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
+-    apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
+-    {{- end }}
++{{- if .Values.rbac.pspAnnotations }}
++    annotations: {{ toYaml .Values.rbac.pspAnnotations | nindent 4 }}
++{{- end }}
+ spec:
+   privileged: false
+   allowPrivilegeEscalation: false
 diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/charts/grafana/values.yaml packages/rancher-monitoring/charts/charts/grafana/values.yaml
 --- packages/rancher-monitoring/charts-original/charts/grafana/values.yaml
 +++ packages/rancher-monitoring/charts/charts/grafana/values.yaml
-@@ -1,7 +1,7 @@
+@@ -1,7 +1,17 @@
  rbac:
    create: true
    pspEnabled: true
 -  pspUseAppArmor: true
-+  pspUseAppArmor: false
++  pspAnnotations: {}
++  ## Specify pod annotations
++  ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#apparmor
++  ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#seccomp
++  ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#sysctl
++  ##
++  # seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
++  # seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
++  # apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
++  # apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
```

Comment:
I don't follow the purpose of this section?

Reply:
Based on the issue logged by @jiaqiluo in https://github.com/rancher/rancher/issues/28536#issuecomment-696876105, it seems like these annotations aren't necessarily supported by all hardened clusters. So instead of using the original Grafana chart approach of [introducing flags like `rbac.pspUseAppArmor`](https://github.com/rancher/charts/pull/680/files#diff-ed5b5b87cf149adc6a6408bf81993efdR203-R209) to opt-out of certain annotations, I used the `kube-prometheus-stack` approach of allowing the user to [opt into annotations via `rbac.pspAnnotations`](https://github.com/prometheus-community/helm-charts/blob/a2b924bf7c439f0531634983dd941eeec2917725/charts/kube-prometheus-stack/values.yaml#L92-L100).

These commented out annotations are just the default ones that the upstream Grafana chart normally adds onto the PSP annotations.

Reply:
Can we drop them if they aren't needed? The patch file is already big without things that we don't need/use

Reply:
We could, but it’s possible / likely a user might want to uncomment these annotations if they have seccomp or apparmor in their cluster. Also it wouldn’t make much of a difference in the patch file since we’ll have to apply a patch anyways to replace pspUseAppArmor with pspAnnotations.